projects/dotcms-field-elements/src/components/dot-form/dot-form.e2e.ts

```diff
@@ -1,127 +1,305 @@
+            });
         });
     });
 
-    it('should listen for statusChange, change form classes and invalidate Submit button', async () => {
-        const textField = await page.find('dot-textfield');
-        const newValue = {
-            name: 'field1',
-            status: {
-                dotPristine: false,
-                dotTouched: false,
-                dotValid: false
-            }
-        };
-
-        textField.triggerEvent('statusChange', {
-            bubbles: true,
-            cancelable: false,
-            detail: newValue
+    describe('buttons', () => {
+        describe('submit', () => {
+            it('should have type submit', async () => {
+                const button = await getSubmitButton();
+                expect(button.getAttribute('type')).toBe('submit');
+            });
         });
 
-        // tslint:disable-next-line:max-line-length
-        const formStatusExpectedMarkup = `<dot-form submit-label=\"Saved\" reset-label=\"Reseted\" class=\"dot-untouched hydrated dot-invalid dot-dirty\"><form><dot-textfield class=\"dot-valid dot-pristine dot-untouched dot-required hydrated\"><div class=\"dot-field__label\"><label for=\"dot-field1\">field1</label><span class=\"dot-field__required-mark\">*</span></div><input id=\"dot-field1\" placeholder=\"\" required=\"\" type=\"text\"><span class=\"dot-field__hint\">hint1</span></dot-textfield><dot-textfield class=\"dot-valid dot-pristine dot-untouched hydrated\"><div class=\"dot-field__label\"><label for=\"dot-field2\">field2</label></div><input id=\"dot-field2\" placeholder=\"\" type=\"text\"><span class=\"dot-field__hint\">hint2</span></dot-textfield><div class=\"form__buttons\"><button type=\"button\">Reseted</button><button type=\"submit\" disabled=\"\">Saved</button></div></form></dot-form>`;
-        await page.waitForChanges();
-        expect(element.outerHTML).toBe(formStatusExpectedMarkup);
+        describe('reset', () => {
+            it('should have type reset', async () => {
+                const button = await getResetButton();
+                expect(button.getAttribute('type')).toBe('reset');
+            });
+        });
     });
 
-    it('should reset event', async () => {
-        const resetBtn = await element.find('button[type="button"]');
-        const expectedStatus = Object.assign({}, formStatus);
-        Object.keys(expectedStatus).forEach(e => expectedStatus[e] = '');
+    describe('actions', () => {
+        beforeEach(async () => {
+            element.setProperty('fields', fieldMockNotRequired);
+            await page.waitForChanges();
+        });
+
+        describe('click reset button', () => {
+            it('should empty values', async () => {
+                await fillTextfield('hello world');
+                await page.waitForChanges();
+                const [textfield, keyvalue, select] = await getFields();
+
+                expect(await textfield.getProperty('value')).toBe('hello world');
```

Comment:
should not populate the other two fields ?

Reply:
No need, they are popuplated by the default values in the mock, I'm just doing this to make sure that I can modify the form before resetting it.